src/vs/workbench/contrib/files/browser/views/explorerDecorationsProvider.ts

```diff
@@ -59,6 +61,9 @@ export class ExplorerDecorationsProvider implements IDecorationsProvider {
 		this.toDispose.add(explorerRootErrorEmitter.event((resource => {
 			this._onDidChange.fire([resource]);
 		})));
+		this.toDispose.add(fileService.onDidFilesChange(e => {
```

Comment:
This event can fire thousands of times at the same time for thousands of files, so whoever is listening needs to be checked for perf bottlenecks if that happens.